File: tests/mv_dots_afterTest.php
<?php

include_once __DIR__ . '/../../src/include_files.php';

use PHPUnit\Framework\TestCase;
use function WpRefs\MovesDots\move_dots_after_refs;

class mv_dots_afterTest extends TestCase
{
    private function assertEqualCompare(string $expected, string $input, string $result)
    {
        if ($result === $input && $result !== $expected) {
            $this->fail("No changes were made! The function returned the input unchanged:\n$result");
        } else {
            $this->assertEquals($expected, $result, "Unexpected result:\n$result");
        }
    }
    // Tests for move_dots_after_refs function
    public function testMoveDotsAfterSingleDot()
    {
        $input = "This is a sentence。<ref>Reference 1</ref>";
        $expected = "This is a sentence<ref>Reference 1</ref>。";
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'en'));
    }

    public function testMoveDotsAfterMultipleDots()
    {
        $input = "First sentence. Second sentence.<ref>Reference 1</ref>";
        $expected = "First sentence. Second sentence<ref>Reference 1</ref>.";
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'en'));
    }

    public function testMoveDotsAfterMultipleRefs()
    {
        $input = "Text।<ref>Ref1</ref><ref>Ref2</ref>";
        $expected = "Text<ref>Ref1</ref><ref>Ref2</ref>।";
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'en'));
    }

    public function testMoveDotsAfterNoDot()
    {
        $input = "Text<ref>Reference</ref>";
        $expected = "Text<ref>Reference</ref>";
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'en'));
    }

    public function testMoveDotsAfterDifferentPunctuation()
    {
        $input = "Text, <ref>Reference</ref>";
        $expected = "Text<ref>Reference</ref>,";
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'en'));
    }

    public function testMoveDotsAfterHy()
    {
        $input = 'Հետծննդյան հոգեբանական խանգարումը հանդիպում է 1000 ծննդաբերությունից 1-2-ի մոտ։ <ref name="Os2018" /><ref name="Li2018" /> Տարբեր [[Մշակույթ|մշակույթներում]] և [[Դասակարգային կառուցվածք|սոցիալական դասերում]] գները նման են թվում։ <ref name="Luc2021" /> Ավելի հաճախ այն հանդիպում է հայտնի կամ նոր սկսվող երկբևեռ խանգարման համատեքստում, որը հայտնի է որպես հետծննդյան երկբևեռ խանգարում : <ref name="Luc2021" /> Այս վիճակը նկարագրվել է դեռևս մ.թ.ա. 400 թվականից [[Հիպոկրատ|Հիպոկրատի]] կողմից ։ <ref name="Os2018" />';
        $expected = 'Հետծննդյան հոգեբանական խանգարումը հանդիպում է 1000 ծննդաբերությունից 1-2-ի մոտ<ref name="Os2018" /><ref name="Li2018" />։ Տարբեր [[Մշակույթ|մշակույթներում]] և [[Դասակարգային կառուցվածք|սոցիալական դասերում]] գները նման են թվում<ref name="Luc2021" />։ Ավելի հաճախ այն հանդիպում է հայտնի կամ նոր սկսվող երկբևեռ խանգարման համատեքստում, որը հայտնի է որպես հետծննդյան երկբևեռ խանգարում : <ref name="Luc2021" /> Այս վիճակը նկարագրվել է դեռևս մ.թ.ա. 400 թվականից [[Հիպոկրատ|Հիպոկրատի]] կողմից<ref name="Os2018" />։';
        $this->assertEqualCompare($expected, $input, move_dots_after_refs($input, 'hy'));
    }
}

File: tests/mv_dots_afterTest.py
from __future__ import annotations

from typing import Iterable, Tuple

import pytest

from src.helps_bots.mv_dots import move_dots_after_refs
from tests.conftest import assert_equal_compare


def _cases() -> Iterable[Tuple[str, str, str]]:
    yield "This is a sentence。<ref>Reference 1</ref>", "This is a sentence<ref>Reference 1</ref>。", "en"
    yield "First sentence. Second sentence.<ref>Reference 1</ref>", "First sentence. Second sentence<ref>Reference 1</ref>.", "en"
    yield "Text।<ref>Ref1</ref><ref>Ref2</ref>", "Text<ref>Ref1</ref><ref>Ref2</ref>।", "en"
    yield "Text<ref>Reference</ref>", "Text<ref>Reference</ref>", "en"
    yield "Text, <ref>Reference</ref>", "Text<ref>Reference</ref>,", "en"
    yield "Text.  <ref>Reference</ref>", "Text<ref>Reference</ref>.", "en"
    yield "Text.<ref name=\"ref1\" />", "Text<ref name=\"ref1\" />.", "en"
    yield "Text. <ref>Ref1</ref> <ref>Ref2</ref>", "Text<ref>Ref1</ref> <ref>Ref2</ref>.", "en"
    yield "This is a sentence. This is another sentence<ref>Reference</ref>", "This is a sentence. This is another sentence<ref>Reference</ref>", "en"
    yield "Text.,<ref>Reference</ref>", "Text<ref>Reference</ref>.,", "en"
    yield "", "", "en"
    yield "This is a sentence.", "This is a sentence.", "en"
    yield "Text.<ref name=\"ref1\" group=\"group1\">Reference content</ref>", "Text<ref name=\"ref1\" group=\"group1\">Reference content</ref>.", "en"
    yield "Text.<ref>Reference with <i>italic</i> text</ref>", "Text<ref>Reference with <i>italic</i> text</ref>.", "en"
    yield "这是句子。<ref>参考文献1</ref>", "这是句子<ref>参考文献1</ref>。", "en"
    yield "यह वाक्य है।<ref>संदर्भ 1</ref>", "यह वाक्य है<ref>संदर्भ 1</ref>।", "en"
    yield "Տեքստ.,<ref>Հղում</ref>", "Տեքստ<ref>Հղում</ref>.,", "hy"
    yield "Text.<ref>Reference</ref>", "Text<ref>Reference</ref>.", "en"
    yield "First sentence.<ref>Ref1</ref> Second sentence.<ref>Ref2</ref>", "First sentence<ref>Ref1</ref>. Second sentence<ref>Ref2</ref>.", "en"
    yield ".<ref>Reference</ref>", "<ref>Reference</ref>.", "en"
    yield "Text. <ref>Reference</ref>", "Text<ref>Reference</ref>.", "en"
    yield "Text, <ref>Reference</ref>", "Text<ref>Reference</ref>,", "en"
    yield "Text.<ref>Reference</ref>", "Text<ref>Reference</ref>.", "en"
    yield "Text,<ref>Reference</ref>", "Text<ref>Reference</ref>,", "en"
    yield "Text.<ref>Ref1</ref><ref>Ref2</ref>", "Text<ref>Ref1</ref><ref>Ref2</ref>.", "en"
    yield "Text. <ref>Ref1</ref> <ref>Ref2</ref>", "Text<ref>Ref1</ref> <ref>Ref2</ref>.", "en"


@pytest.mark.parametrize("input_text, expected, lang", list(_cases()))
def test_move_dots_after_cases(input_text: str, expected: str, lang: str) -> None:
    assert_equal_compare(expected, input_text, move_dots_after_refs(input_text, lang))
